src/panels/lovelace/cards/hui-history-graph-card.js

```diff
@@ -41,21 +44,55 @@ class HuiHistoryGraphCard extends PolymerElement {
     return {
       hass: Object,
       _config: Object,
-      stateHistory: Object,
+      stateHistory: {
+        type: Object,
+        observer: '_historyChanged'
+      },
       stateHistoryLoading: Boolean,
     };
   }
 
+  _historyChanged() {
+    if (!this.stateHistory) {
+      return;
+    }
+
+    // eslint-disable-next-line
+    for (const entity of this.stateHistory.timeline) {
+      if (this._names[entity.entity_id]) {
+        entity.name = this._names[entity.entity_id];
```

Comment:
We shouldn't change data that we don't own. This adds a coupling to `stateHistory` and it can no longer change how it works without also potentially breaking this component.

Instead an override name option should be added to `state-history-charts`

Reply:
No. The data component fetches the data and that should be untouched. Instead, while rendering the chart we should override the name.

Reply:
So the code in general is ok but should be moved to state-history-charts?

Reply:
The object is ok. The code is not because it mutates data that is generated by `ha-state-history-data`. Instead, the rendering of charts should use the data and an override object to show different names in the UI.

Reply:
Ok, it's done. And yes, there's no underscore but I consider it a Polymer property which don't have underscores (hass, config, stateObj, etc.)

Could you test with regular ui? I'm not sure if names is null or {} there.